packages/material-ui-lab/src/Autocomplete/Autocomplete.js

```diff
@@ -122,6 +122,9 @@ export const styles = (theme) => ({
         padding: '2.5px 4px',
       },
     },
+    '& [class*="MuiInputAdornment-filled"][class*="MuiInputAdornment-positionStart"]:not([class*="MuiInputAdornment-hiddenLabel"])': {
```

Comment:
This won't work with nested themes or any themes that use a different class name generator, no?

Reply:
As far as I know, it works because the nested themes only add a suffix.

Reply:
What about custom class name generators?

Reply:
Will be solved in v5 as we are moving to hard-coded class names.

Reply:
Mind that we are already using the `&[class*="` 6 times on this file (prior to the change)

Reply:
> If it's a blocker for some teams, they will tell us :)

I'm not arguing for or against that future. After all, this feature existed before I joined and it's likely that you introduced it so you should know the best why you added it.

Though right now we still have that capability and if we no longer support it we should first make sure to strip that feature out completely.

> Mind that we are using the &[class*=" pattern 6 times on this file (prior to the changes)

An existing bug does not justify adding more.

Reply:
I don't think that it's fair to qualify this a "new" bug. Solving this occurrence will require the exact same solution as solving the other. I think that it's an area of improvement but I don't think that it's directly related to #22544. I think that it surfaces either that they should be a prop to inverse the padding container (Autocomplete has the same requirement or similar to textarea) or #16113

Reply:
Actually, we probably want to fix the textarea at the same time: https://github.com/mui-org/material-ui/issues/22544#issuecomment-691041009